core/util.go

```diff
@@ -108,30 +107,47 @@ func ServeHTTPGatewayProxy(node *TextileNode) (<-chan error, error) {
 		w.Write(b)
 	})
 
-	// get config and set proxy address to raw gateway address plus one,
-	// so a gateway on 8182 means the proxy will run on 9182
-	cfg, err := node.Context.GetConfig()
+	errc := make(chan error)
+
+	port, err := node.GatewayPort()
 	if err != nil {
-		return nil, fmt.Errorf("ServeHTTPGatewayProxy: GetConfig() failed: %s", err)
+		return errc, err
 	}
-	tmp := strings.Split(cfg.Addresses.Gateway, "/")
-	gaddrs := tmp[len(tmp)-1]
-	gaddr, err := strconv.ParseInt(gaddrs, 10, 64)
+
+	portString := fmt.Sprintf(":%d", port)
+	fmt.Println(portString)
```

Comment:
log.Debugf if needed. you can `tail -f .ipfs/logs/textile.log` to get a much better debug experience than just printing to stdout

Reply:
not needed, left-over from dev